packages/node_modules/@node-red/nodes/core/common/lib/debug/debug-utils.js

```diff
@@ -233,6 +233,35 @@ RED.debug = (function() {
 
     }
 
+    function containsDebug(sid, map) {
+        if (sid in map) {
+            return true;
+        }
+        var nodes = RED.nodes.filterNodes({z: sid});
+        var contain = false;
+        nodes.forEach(function (n) {
```

Comment:
It would be more efficient to use a `for` loop here to allow the function to return as soon as it finds a match. With this function, it has to loop over every node every single time, even if the very first node returned by filterNodes is a match.